sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosClientCache.scala

```diff
@@ -152,13 +154,16 @@ private[spark] object CosmosClientCache extends BasicLoggingTrait {
             case None => customApplicationNameSuffix
           }
 
-          builder.clientTelemetryConfig().clientMetrics(CosmosClientMetrics.meterRegistry.get)
-          builder.clientTelemetryConfig().clientCorrelationId(clientCorrelationId)
-          builder.clientTelemetryConfig().metricTagNames(
+          val telemetryConfig  = new CosmosClientTelemetryConfig()
+          telemetryConfig.clientMetrics(CosmosClientMetrics.meterRegistry.get)
+          telemetryConfig.clientCorrelationId(clientCorrelationId)
```

Comment:
it'd be nice to use `set` and `get` verbs when we have both. We skip `set` when there is no getter through.

Reply:
All getters are internal.